packages/ui/src/components/templates/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,153 @@
+}
+
+interface Line {
+  graphList: Graph[];
+}
+
+interface DashboardProps {
+  style?: any;
+  lineList: Line[];
+}
+
+const styleGraph = (nbGraphInLine: number, type: 'style' | 'width') => {
+  let style = null;
+  let width = null;
+
+  if (nbGraphInLine === 1) {
+    style = {width: Dimensions.get('window').width - 60};
+    width = Dimensions.get('window').width - 160;
+  } else if (Dimensions.get('window').width < 500 && nbGraphInLine !== 1) {
+    style = {width: Dimensions.get('window').width / 2 - 15};
+    width = Dimensions.get('window').width / 4;
+  } else if (nbGraphInLine === 2) {
+    style = {width: Dimensions.get('window').width / 2 - 60};
+    width = Dimensions.get('window').width / 3 - 60;
+  } else if (nbGraphInLine === 3) {
+    style = {width: Dimensions.get('window').width / 3 - 60};
+    width = Dimensions.get('window').width / 5 - 60;
+  } else if (nbGraphInLine === 4) {
+    style = {width: Dimensions.get('window').width / 4 - 20};
+    width = Dimensions.get('window').width / 6 - 60;
+  }
+
+  return type === 'style' ? style : width;
+};
+
+const LineChartDashboardRender = (datasets, key, nbGraphInLine, title) => {
+  return (
+    <LineChartDashboard
+      datasets={datasets}
+      key={key}
+      style={styleGraph(nbGraphInLine, 'style')}
+      widthGraph={styleGraph(nbGraphInLine, 'width')}
+      title={title}
+    />
+  );
+};
+
+const PieChartRender = (datasets, key) => {
+  return <PieChartDashboard datasets={datasets} key={key} legend={true} />;
+};
+
+const BarChartDashboardRender = (datasets, key, nbGraphInLine, title) => {
+  return (
+    <BarChartDashboard
+      datasets={datasets}
+      key={key}
+      style={styleGraph(nbGraphInLine, 'style')}
+      widthGraph={styleGraph(nbGraphInLine, 'width')}
+      title={title}
+    />
+  );
+};
+
+const Dashboard = ({style, lineList}: DashboardProps) => {
+  return (
+    <ScrollView style={[styles.container, style]}>
+      {lineList?.map((line, indexLine) => {
+        const nbGraphInLine =
+          line.graphList?.length > 4 ? 4 : line.graphList?.length;
+        return (
+          <View style={styles.lineContainer} key={indexLine}>
+            {line?.graphList.map((graph, indexGraph) => {
+              const title = graph?.title;
+              if (indexGraph > 4) {
+                return null;
+              }
+              if (graph?.type === 'bar') {
+                return BarChartDashboardRender(
+                  graph.dataList,
+                  indexGraph,
+                  nbGraphInLine,
+                  title,
+                );
+              }
+              if (graph?.type === 'pie') {
+                return PieChartRender(graph.dataList, indexGraph);
+              }
+              if (graph?.type === 'line') {
+                return LineChartDashboardRender(
+                  graph.dataList,
+                  indexGraph,
+                  nbGraphInLine,
+                  title,
+                );
+              }
```

Comment:
Create a render function return the right component depending on type and then give the props to the given component.
Moreover, graph type should be define somewhere and not be hard written in conditions.